openmc/surface.py

```diff
@@ -94,6 +94,22 @@ def __repr__(self):
 
         return string
 
+    def __key(self):
+        return (self.type, self.boundary_type) + \
+                tuple(j for i in self.coefficients.items() for j in i)
```

Comment:
For this method, we want the order of the coefficients to be consistent, which is not ensured with dictionaries unless care is taken. Thankfully, when Surfaces are constructed, the coefficients dictionary is populated in some deterministic manner ([Sample from `Cone` base class](https://github.com/openmc-dev/openmc/blob/55ca9fe7f81e271e6167aea98677f2d2660eee6b/openmc/surface.py#L1624-L1627)), which can ensure insertion order and iteration as well.  But it may be possible that someone populates this dictionary arbitrarily.
```python
>>> foo = {"x": 2, "y": 4}
>>> bar = {"y": 4, "x": 2}
```
If these dictionaries represented the coefficients for identical surface types, they _should_ hash as identical, but due to the dictionary ordering will be hashed differently. 
```python
>>> list(foo.items())
[("x", 2), ("y", 4)]
>>> list(bar.items())
[("y", 4), ("x", 2)]
```
For robustness, I would recommend using a sorted iteration
```python
>>> list(sorted(bar.items()))
[("x", 2), ("y", 4)]
```

Reply:
Good point, thanks.